pkg/client/apiutil/restmapper.go

```diff
@@ -75,6 +75,12 @@ func (m *mapper) KindFor(resource schema.GroupVersionResource) (schema.GroupVers
 
 // KindsFor implements Mapper.KindsFor.
 func (m *mapper) KindsFor(resource schema.GroupVersionResource) ([]schema.GroupVersionKind, error) {
+	if resource.Version == "" {
+		if err := m.addKnownGroupAndReload(resource.Group, resource.Version); err != nil {
```

Comment:
Isn't this a bit over eager and removing the "lazy" aspect if we always reload as soon as we get a request without version?

Reply:
Yeah, can be a bit over eager, but is it possible to fetch less and at the same time return the complete list?

Or is this by design, and https://github.com/kubernetes-sigs/controller-runtime/issues/2349 not actually a bug?